src/nanotron/data/dataset_builder.py

```diff
@@ -0,0 +1,209 @@
+from typing import Any, List, Type, Union
+
+import numpy
+import torch
+
+from nanotron import logging
+from nanotron.data.blended_nanoset import BlendedNanoset
+from nanotron.data.indexed_dataset import MMapIndexedDataset
+from nanotron.data.nanoset import Nanoset
+from nanotron.data.nanoset_configs import NanosetConfig
+from nanotron.data.utils import Split, log_BlendedNanoset_stats, normalize
+
+logger = logging.get_logger(__name__)
+
+DistributedDataset = Union[Nanoset, MMapIndexedDataset, BlendedNanoset]
+
+
+class NanosetBuilder(object):
+    """Builder class for the Nanoset classes
+
+    Args:
+
+        config (NanosetConfig): The config object which informs dataset creation
+    """
+
+    def __init__(
+        self,
+        config: NanosetConfig,
+    ):
+        self.config = config
+
+    def build(self) -> List[Union[BlendedNanoset, Nanoset]]:
```

Comment:
is there a reason why we have a `build()` function? Maybe we can just return the nanosets in the init call?

Reply:
`NanosetBuilder` returns a list of `Nanosets` or `BlendedNanosets` according to the `NanosetConfig`. I know it's not the most aesthetic solution, but in `__init__`, we can't return anything, so we use `__init__` to set the object's attributes and `build()` to return the constructed datasets. I tried using other methods like `__new__`, and it ended up being more complex.